pkg/operator/rgw/rgw.go

```diff
@@ -89,8 +110,141 @@ func (c *Cluster) Start() error {
+	updatePeriod := false
+
+	// create the realm if it doesn't exist yet
+	output, err := c.runRGWCommand("realm", "get", realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("realm", "create", realmArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw realm %s. %+v", c.Config.Name, err)
+		}
+	}
+
+	realmID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse realm id. %+v", err)
+	}
+
+	// create the zonegroup if it doesn't exist yet
+	output, err = c.runRGWCommand("zonegroup", "get", zonegroupArg, realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("zonegroup", "create", "--master", zonegroupArg, realmArg, endpointArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw zonegroup for %s. %+v", c.Config.Name, err)
+		}
+	}
+
+	zoneGroupID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse realm id. %+v", err)
+	}
+
+	// create the zone if it doesn't exist yet
+	output, err = c.runRGWCommand("zone", "get", zoneArg, zonegroupArg, realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("zone", "create", "--master", endpointArg, zoneArg, zonegroupArg, realmArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw zonegroup for %s. %+v", c.Config.Name, err)
+		}
+	}
+	zoneID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse zone id. %+v", err)
+	}
+
+	if updatePeriod {
+		_, err = c.runRGWCommand("period", "update", "--commit")
+		if err != nil {
+			return fmt.Errorf("failed to update period. %+v", err)
+		}
+	}
+
+	logger.Infof("RGW: realm=%s, zonegroup=%s, zone=%s", realmID, zoneGroupID, zoneID)
+	return nil
+}
+
+func (c *Cluster) createPools() error {
+	metadataPools := []string{
+		".rgw.root",
+		"rgw.control",
+		"rgw.meta",
+		"rgw.log",
+		"rgw.buckets.index",
+	}
+	if err := c.createSimilarPools(metadataPools, c.Config.MetadataConfig); err != nil {
+		return fmt.Errorf("failed to create metadata pools. %+v", err)
+	}
+
+	dataPools := []string{"rgw.buckets.data"}
+	if err := c.createSimilarPools(dataPools, c.Config.DataConfig); err != nil {
+		return fmt.Errorf("failed to create data pool. %+v", err)
+	}
+
+	return nil
+}
+
+func (c *Cluster) createSimilarPools(pools []string, poolConfig model.Pool) error {
+	cephConfig := ceph.ModelPoolToCephPool(poolConfig)
+	if cephConfig.ErasureCodeProfile != "" {
+		// create a new erasure code profile for the new pool
+		if err := ceph.CreateErasureCodeProfile(c.context, c.Namespace, poolConfig.ErasureCodedConfig, cephConfig.ErasureCodeProfile); err != nil {
+			return fmt.Errorf("failed to create erasure code profile for object store %s: %+v", c.Config.Name, err)
+		}
+	}
+
+	for _, pool := range pools {
+		// create the pool if it doesn't exist yet
+		name := pool
+		if !strings.HasPrefix(pool, ".") {
+			name = fmt.Sprintf("%s.%s", c.Config.Name, pool)
```

Comment:
it looks like only pools that start with `.` will get the object store name embedded into the pool name.  does that mean that the other pools are shared across object store instances?

Reply:
right, there is a single pool named `.rgw.root` that tracks across all object stores. I'll add a comment

Reply:
ah oops, i missed the `!`, you're right that it's just the sole `.rgw.root` that is shared across object stores.  sounds good.